packages/fluentui/react-builder/src/narration/NarrationComputer.tsx

```diff
@@ -23,14 +35,186 @@ export class NarrationComputer {
     usage: '',
   };
 
-  // Computes and returns the screen reader narration for the given element and platform.
-  async compute(element: Element, platform: string): Promise<string> {
+  // Traverses the DOM rooted at the given element to find all the focusable elements and returns them.
+  async getFocusableElements(element: IAriaElement): Promise<FocusableElement[]> {
+    // Prepare all the arrays
+    const activeDescendantsParents: IAriaElement[] = [];
+    const path: string[] = [];
+    const focusableElements: FocusableElement[] = [];
+
+    // Traverse down the DOM tree rooted at the element to first find the activeDescendants parent elements, and then all the focusable elements and their paths
+    this.findActiveDescendantsParents(element, activeDescendantsParents);
+    await this.findFocusableElements(element, path, focusableElements, activeDescendantsParents);
+
+    return focusableElements;
+  } // End getFocusableElements
+
+  // Recursively traverses the DOM tree rooted at the given element to find all the parents which have the ariaActiveDescendantElement property set, and saves the found elements to the given parents array.
+  findActiveDescendantsParents(element: IAriaElement, parents: IAriaElement[]) {
+    if (element.ariaActiveDescendantElement != null) {
+      // Begin if 1
+      parents.push(element);
+    } // End if 1
+    for (let i = 0; i < element.children.length; i++) {
```

Comment:
`.forEach` usage?

Reply:
Again, HTMLCollection has no forEach() method.

Reply:
I have replaced it with Array.from().